benches/hierarchy/mvcc/mod.rs

```diff
@@ -0,0 +1,162 @@
+            ).generate(DEFAULT_ITERATIONS);
+            for (k, v) in &kvs {
+                txn.prewrite(
+                    Mutation::Put((Key::from_raw(&k), v.clone())),
+                    &k.clone(),
+                    &option,
+                ).unwrap();
+            }
+            let modifies = txn.into_modifies();
+            let _ = engine.async_write(&ctx, modifies, Box::new(move |(_, _)| {}));
+            let keys: Vec<Key> = kvs.iter().map(|(k, _)| Key::from_raw(&k)).collect();
+            let snapshot = engine.snapshot(&ctx).unwrap();
+            (snapshot, keys)
+        },
+        |(snapshot, keys)| {
+            for key in keys {
+                let mut txn = MvccTxn::new(snapshot.clone(), 1, false).unwrap();
+                black_box(txn.commit(key, 1)).unwrap();
+            }
+        },
+    );
+}
+
+fn mvcc_reader_load_lock<E: Engine, F: EngineFactory<E>>(b: &mut Bencher, config: &BenchConfig<F>) {
+    let engine = config.engine_factory.build();
+    let ctx = Context::default();
+    let test_keys: Vec<Key> = KvGenerator::with_seed(
+        config.key_length,
+        config.value_length,
+        DEFAULT_KV_GENERATOR_SEED,
+    ).generate(DEFAULT_ITERATIONS)
+        .iter()
+        .map(|(k, _)| Key::from_raw(&k))
+        .collect();
+
+    b.iter_with_setup(
+        || {
+            let snapshot = engine.snapshot(&ctx).unwrap();
+            (snapshot, &test_keys)
+        },
+        |(snapshot, test_kvs)| {
+            for key in test_kvs {
+                let mut reader = MvccReader::new(
+                    snapshot.clone(),
+                    None,
+                    false,
+                    None,
+                    None,
+                    ctx.isolation_level,
+                );
+                black_box(reader.load_lock(&key).unwrap());
+            }
+        },
+    );
+}
+
+fn mvcc_reader_seek_write<E: Engine, F: EngineFactory<E>>(
+    b: &mut Bencher,
+    config: &BenchConfig<F>,
+) {
+    let engine = config.engine_factory.build();
+    let ctx = Context::default();
+    b.iter_with_setup(
+        || {
+            let snapshot = engine.snapshot(&ctx).unwrap();
+            let test_keys: Vec<Key> = KvGenerator::with_seed(
+                config.key_length,
+                config.value_length,
+                DEFAULT_KV_GENERATOR_SEED,
+            ).generate(DEFAULT_ITERATIONS)
+                .iter()
+                .map(|(k, _)| Key::from_raw(&k))
+                .collect();
+            (snapshot, test_keys)
+        },
+        |(snapshot, test_keys)| {
+            for key in &test_keys {
+                let mut reader = MvccReader::new(
+                    snapshot.clone(),
+                    None,
+                    false,
+                    None,
+                    None,
+                    ctx.isolation_level,
+                );
+                black_box(reader.seek_write(&key, u64::max_value()).unwrap());
```

Comment:
seek_write is an interface to be removed. I think it is not necessary to bench it here.

Reply:
Ok, I will remove it later.